framework/doc/content/source/kokkos/functions/KokkosPiecewiseConstant.md

```diff
@@ -0,0 +1,24 @@
+# KokkosPiecewiseConstant
+
+!if! function=hasCapability('kokkos')
+
+This is the Kokkos version of [PiecewiseConstant](PiecewiseConstant.md). See the original document for details.
+
+## Example Syntax
+
+!listing test/tests/kokkos/functions/piecewise_constant/kokkos_piecewise_constant.i start=[func_x_y] end=[] include-end=true
+
+!listing test/tests/kokkos/functions/piecewise_constant/kokkos_piecewise_constant.i start=[func_xy_data] end=[] include-end=true
+
+!listing test/tests/kokkos/functions/piecewise_constant/kokkos_piecewise_constant.i start=[func_csv] end=[] include-end=true
+
+!listing test/tests/kokkos/functions/piecewise_constant/kokkos_piecewise_constant.i start=[func_json] end=[] include-end=true
```

Comment:
```suggestion
!listing test/tests/kokkos/functions/piecewise_constant/kokkos_piecewise_constant.i block=KokkosFunctions
```
I've grown suspicious that we're missing some doc hooks?

Reply:
Did you just not want to show the top-level block before? I was worried that there was a documentation issue if you used the syntax I suggested. But the fact that you made the changes I suggested makes me not worry